erpnext/payroll/doctype/salary_structure/test_salary_structure.py

```diff
@@ -139,10 +140,23 @@ def make_salary_structure(salary_structure, payroll_frequency, employee=None, do
 	else:
 		salary_structure_doc = frappe.get_doc("Salary Structure", salary_structure)
 
+	filters = {'employee':employee, 'docstatus': 1}
+	if not from_date and payroll_period:
+		from_date = payroll_period.start_date
```

Comment:
This code has been added because after newly introduced validation, a Salary Structure Assignment must be created on or before the Salary Slip date. This code assumes that employee has joined before Payroll Period. Alternatively, `from_date` can be manually passed to `make_salary_structure` in these cases.